docs_user/modules/proc_deploying-a-ceph-ingress-daemon.adoc

```diff
@@ -0,0 +1,131 @@
+----
+
+. Set the required images for both HAProxy and Keepalived: 
++ 
+----
+ifeval::["{build}" != "downstream"]
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_haproxy quay.io/ceph/haproxy:2.3
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_keepalived quay.io/ceph/keepalived:2.1.5
+endif::[]
+ifeval::["{build}" == "downstream"]
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_haproxy registry.redhat.io/rhceph/rhceph-haproxy-rhel9:latest
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_keepalived registry.redhat.io/rhceph/keepalived-rhel9:latest
+endif::[]
+----
+
+. Prepare the ingress spec and mount it to cephadm:
++
+----
+$ sudo vim /home/ceph-admin/specs/rgw_ingress
+----
+
+. Copy and paste the following content in to the ingress spec:
+//kgilliga: Is this accurate? The user is pasting the content in to the ingress spec?
++
+[source,yaml]
+----
+---
+service_type: ingress
+service_id: rgw.rgw
+placement:
+  label: rgw
+spec:
+  backend_service: rgw.rgw
+  virtual_ip: 10.0.0.89/24
+  frontend_port: 8080
+  monitor_port: 8898
+  virtual_interface_networks:
+    - 10.0.0.0/24
+----
+
+. Mount the generated spec and apply it by using the orchestrator CLI:
++
+----
+$ cephadm shell -m /home/ceph-admin/specs/rgw_ingress
+$ cephadm shell -- ceph orch apply -i /mnt/rgw_ingress
+----
+
+. Wait until the ingress is deployed and query the resulting endpoint:
++
+----
+[ceph: root@controller-0 /]# ceph orch ls
+
+NAME                 	PORTS            	RUNNING  REFRESHED  AGE  PLACEMENT
+crash                                         	6/6  6m ago 	3d   *
+ingress.rgw.rgw      	10.0.0.89:8080,8898  	6/6  37s ago	60s  label:rgw
+mds.mds                   3/3  6m ago 	3d   controller-0;controller-1;controller-2
+mgr                       3/3  6m ago 	3d   controller-0;controller-1;controller-2
+mon                       3/3  6m ago 	3d   controller-0;controller-1;controller-2
+osd.default_drive_group   15  37s ago	3d   cephstorage-0;cephstorage-1;cephstorage-2
+rgw.rgw   ?:8090          3/3  37s ago	4m   label:rgw
+----
++
+----
+[ceph: root@controller-0 /]# curl  10.0.0.89:8080
+
+---
+<?xml version="1.0" encoding="UTF-8"?><ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Owner><ID>anonymous</ID><DisplayName></DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>[ceph: root@controller-0 /]#
+—
+----
++
+The result shows that you can reach the backend from the
+ingress daemon. You are almost ready to interact with it using the
+swift CLI.
+//kgilliga: Do we want to say "You are almost ready to interact with it using the swift CLI" in the downstream docs?
```

Comment:
You can change `Swift` to the downstream name, but the idea is to say that now the migration is completed and users can use RGW through the swift command line.